buildpacks/scala/src/main.rs

```diff
@@ -0,0 +1,415 @@
+            _ => panic!("expected MissingStageTask error"),
+        }
+    }
+
+    #[test]
+    fn check_already_defined_as_error_is_reported() {
+        let stdout = formatdoc! {"
+            [error] Expected ';'
+            [error] Not a valid command: stage (similar: last-grep, set, last)
+            [error] Not a valid project ID: stage
+            [error] Expected ':'
+            [error] Blah is already defined as object Blah
+        "}
+        .into_bytes();
+
+        let output = Output {
+            stdout,
+            stderr: vec![],
+            status: ExitStatus::from_raw(0),
+        };
+        let err = handle_sbt_error(&output);
+        match err {
+            ScalaBuildpackError::AlreadyDefinedAsObject => {}
+            _ => panic!("expected MissingStageTask error"),
+        }
+    }
+}
+
+fn get_sbt_build_tasks(build_config: &BuildConfiguration) -> Vec<String> {
+    let mut tasks: Vec<String> = Vec::new();
+
+    if let Some(true) = &build_config.sbt_clean {
+        tasks.push(String::from("clean"));
+    }
+
+    if let Some(sbt_pre_tasks) = &build_config.sbt_pre_tasks {
+        sbt_pre_tasks
+            .iter()
+            .for_each(|task| tasks.push(task.to_string()));
+    }
+
+    if let Some(sbt_tasks) = &build_config.sbt_tasks {
+        sbt_tasks
+            .iter()
+            .for_each(|task| tasks.push(task.to_string()));
+    } else {
+        let default_tasks = vec![String::from("compile"), String::from("stage")];
+        for default_task in &default_tasks {
+            tasks.push(match &build_config.sbt_project {
+                Some(project) => format!("{project}/{default_task}"),
+                None => default_task.to_string(),
+            });
+        }
+    }
+
+    tasks
+}
+
+#[cfg(test)]
+mod get_sbt_build_tasks_tests {
+    use crate::build_configuration::BuildConfiguration;
+    use crate::get_sbt_build_tasks;
+    use semver::Version;
+
+    #[test]
+    fn get_sbt_build_tasks_with_no_configured_options() {
+        let config = BuildConfiguration {
+            sbt_project: None,
+            sbt_pre_tasks: None,
+            sbt_tasks: None,
+            sbt_clean: None,
+            sbt_opts: None,
+            sbt_available_at_launch: None,
+            sbt_version: Version::new(0, 0, 0),
+        };
+        assert_eq!(get_sbt_build_tasks(&config), vec!["compile", "stage"]);
+    }
+
+    #[test]
+    fn get_sbt_build_tasks_with_all_configured_options() {
+        let config = BuildConfiguration {
+            sbt_project: Some("projectName".to_string()),
+            sbt_pre_tasks: Some(vec!["preTask".to_string()]),
+            sbt_tasks: Some(vec!["task".to_string()]),
+            sbt_clean: Some(true),
+            sbt_opts: None,
+            sbt_available_at_launch: None,
+            sbt_version: Version::new(0, 0, 0),
+        };
+        assert_eq!(
+            get_sbt_build_tasks(&config),
+            vec!["clean", "preTask", "task"]
```

Comment:
Is this correct? Shouldn't `task` be `projectName/task`?

Reply:
that's a tricky one but i'd say no.  if the project name is given then we should append it to the default tasks but if the default tasks are overridden by the user then we shouldn't modify them because that would change what the user gave us.